src/mkdocstrings/handlers/python.py

```diff
@@ -87,7 +90,11 @@ def update_env(self, md: Markdown, config: dict) -> None:  # noqa: D102 (ignore
         self.env.trim_blocks = True
         self.env.lstrip_blocks = True
         self.env.keep_trailing_newline = False
+        self.env.filters["brief_xref"] = self.do_brief_xref
 
+    def do_brief_xref(self, path: str) -> Markup:
+        brief = path.split(".")[-1]
+        return Markup(f"<span data-autorefs-optional=\"{path}\">{brief}</span>")
```

Comment:
Nice way to handle this, but the user has no control over this. I'm willing to merge anyway because I consider it a first step to allow users to shorten paths like that. We can still add some special markup later that will take advantage of this filter, so users can control it.

Reply:
Yeah, it should probably be an option whether to shorten implicit cross-references (or even create them at all).  This seems like a good default, though, especially with the autorefs change to show the full path on hover.